packages/nlp/src/context-manager.js

```diff
@@ -109,8 +108,14 @@ class ContextManager extends Clonable {
           ? this.container.get('database')
           : undefined;
         if (database) {
-          const saved = await database.save(this.settings.tableName, clone);
-          context.id = saved.id;
+          clone.id = id;
+          await database.update(this.settings.tableName, clone, {
```

Comment:
The problem with replacing save with update and options with upsert, is that this will only work if the database is mongoDB. Database object is an ORM so its intended to manage different databases, right now memory and mongodb, but in the future can be others. My suggestion is to keep the save and replace the code of save in the mongodb-adapter to do an upsert

Reply:
ok, I restore to database.save and we'll handle the upsert problem inside.